packages/coinstac-simulator/src/db-server.js

```diff
@@ -5,58 +5,126 @@
+const pdbs = require('spawn-pouchdb-server');
+const Pouchy = require('pouchy');
+const url = require('url');
+
+const Consortium = common.models.Consortium;
+
+/**
+ * Placeholder for the server.
+ *
+ * @type {(undefined|Function)}
+ */
+let server;
 
-const me = {};
+function addDocument(pathname, doc) {
+  const dbUrl = url.format({
+    hostname: 'localhost',
+    pathname,
+    port: config.port,
+    protocol: 'http',
+  });
+
+  const db = new Pouchy({
+    url: dbUrl,
+    sync: 'out',
+  });
+
+  return db.save(doc);
+}
 
 /**
  * @function setup
  * @description boots a pouchdb-server, a dbRegistry instance, and
  * a computation registry instance.  these utilities are commonly
  * required for PipelineRunnerPool testing
+ * @param {string} declPath Path to the declaration file
  * @returns {Promise}
  */
-const setup = () => { // eslint-disable-line
-  return new Promise((res, rej) => {
-    try { cp.execSync('pkill -f pouchdb-server'); } catch (err) {} // eslint-disable-line
-    // @note ^^^ helpful to uncomment when tests are failing
+function setup(declPath) {
+  if (server) {
+    return Promise.reject(new Error('Server already instantiated'));
+  }
+
+  const decl = require(declPath); // eslint-disable-line global-require
 
+  return new Promise((resolve, reject) => {
     // spawn-pouchdb-server mutates user input >:(
     // https://github.com/hoodiehq/spawn-pouchdb-server/pull/33
-    pdbs(cloneDeep(pdbsConfig), (err, srv) => {
-      if (err) { return rej(err); }
-      me.server = srv;
-      return res(srv);
+    pdbs(cloneDeep(config), (error, srv) => {
+      if (error) {
+        reject(error);
+      } else {
+        server = srv;
+        resolve(server);
+      }
     });
-  });
-};
+  })
+    .then(() => {
+      logger.info(`pouchdb-server running on http://localhost:${config.port}`);
+      const users = decl.users.map(u => u.username);
+      const defaultConsortium = new Consortium({
+        _id: `testconsortiumid${Date.now()}`,
+        description: 'test-default-consortium',
+        label: 'test-default-consortium',
+        owners: users,
+        users,
+      });
+      /* eslint-disable global-require */
+      const decentralizedComputation = require(decl.computationPath);
+      /* eslint-enable global-require */
+      const computationDoc = {
+        _id: 'testcomputationid',
+        name: decentralizedComputation.name,
+        version: decentralizedComputation.version,
+      };
+
+      return Promise.all([
+        addDocument('consortia', defaultConsortium.serialize()),
+        addDocument('computations', computationDoc),
+      ]);
+    })
+    .then(responses => {
+      logger.info('Seeded database', {
+        consortium: responses[0]._id,
+        computation: responses[1].name,
+      });
+
+      return server;
+    });
+}
+
+/**
+ * @returns {Promise}
+ */
+function teardown() {
+  if (!server) {
+    return Promise.reject(new Error('Server not instantiated'));
```

Comment:
It was because I was being a dummy in _src/index.js_. Although it makes sense: can’t tear down the server it isn’t yet created.